web/modules/custom/usagov_top_navigation_menu/usagov_top_navigation_menu.module

```diff
@@ -45,3 +46,71 @@ function usagov_top_navigation_menu_get_top_nav() {
   }
   return $topnav;
 }
+
+/**
+ * Implements hook_entity_update() for menu_link_content entity.
+ *
+ * This Hook fires after the MySQL command to save the menu-item as executed.
+ *
+ * We have made this hook in response to USAGOV-1934. We have noticed that in
+ * certain situations, duplicate menu-items are being created where one of
+ * them points to /node/$UUID - we are having hard time reproducing the
+ * problem, but we know it is happening. We are going to catch this edge-case
+ * here and do something.
+ *
+ * See the comments in USAGOV-1934 for more details and how this edge-case
+ * can come about.
+ */
+function usagov_top_navigation_menu_menu_link_content_update($entity) {
+
+  // This function may potentially trigger a save on a menu-item.
+  // We protect against recursive infinity-loops here.
+  static $stopLoop = FALSE;
+  if ($stopLoop === TRUE) {
+    return;
+  }
+
+  // Get the Node ID this menu-item points to
+  $id = $entity->id();
+  $url = $entity->getUrlObject()->toString();
+  $sysPath = \Drupal::service('path_alias.manager')->getPathByAlias($url);
+  if (empty($sysPath)) {
+    return;
+  }
+  $nid = str_replace('/node/', '', $sysPath);
+
+  // Get the UUID for that Node ID
+  $node = Node::load($nid);
+  if (empty($node)) {
+    return;
+  }
+  $uuid = $node->uuid();
+
+  // Check if there is a duplicate menu-item in the system that points to /node/$UUID
+  $db = \Drupal::database();
+  $sql = "SELECT id FROM menu_link_content_data WHERE link__uri LIKE '%node/{$uuid}' AND id <> {$id} ";
+  $results = $db->query($sql)->fetchAllKeyed(0, 0);
+
+  // Disable this menu item
+  $results = [1359];
```

Comment:
Looks like a bit of debugging code left in on this line! It did allow me to see the message cms users will see when this happens. 

I'm just adding this as a comment; I can approve after you remove this line, or someone else can review.

Reply:
Ah, I missed this somehow! This is now removed